validators.py

```diff
@@ -82,10 +82,12 @@ def validate(value: Iterable, **_kwargs) -> None:
     return validate
 
 
-def no_padding_whitespace():
+def no_padding_whitespace_and_no_pipe_character():
```

Comment:
I think these should be separate validators, like this if a value had padding whitespace and a pipe, the pipe won't be spotted.

Also the check for the pipe will be relatively slow over lots of rows, we can be savvy with which columns we apply it too (e.g. column that uses the `in_set` or `numeric` validator doesn't need to also check for the pipe explicitly).